### ISSUE TITLE
Can't select text inside of Rover

### ISSUE THREAD

**Issue by Slapbox** [reakit/reakit#448]
### Current behavior

I'm trying to recreate the behavior of `<Step />` but there's an issue with selecting text rendered inside of a `<Rover />`. It can't be selected.

I tried using `as={Box}` to render it as a `div` but text selection is still not allowed. Using devTools to modify the Rover to be a `div` or `span` works, but only if `as` is not used.

The `focusable` property doesn't change this behavior.

### Steps to reproduce the bug

[Demonstration](https://codesandbox.io/s/reakit-nfu4g)

### Expected behavior

Contents of `<Rover />` should be selectable.

### Possible solutions

Unknown. Issue did not exist in `v0` so perhaps there are some clues there

### Environment

```sh
$ npx envinfo --system --binaries --npmPackages

  System:
    OS: Windows 7
    CPU: (2) x64 Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz
    Memory: 2.98 GB / 13.75 GB
  Binaries:
    Yarn: 1.17.3 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.2.0 - C:\Users\Remie\AppData\Roaming\npm\npm.CMD

```

As far as fully transitioning to `v1` I'd rank this bug as a critical bug for us.

PS:  I'm pretty sure this is the last bug report from me! Besides these few reports everything else works great transitioning from `v0`

**Comment by diegohaz** [comment]
Good catch @Slapbox! Thanks for that!

This happens because `Tabbable` (used by `Rover`) has an `onMouseDown` handler that calls `event.preventDefault()`:
https://github.com/reakit/reakit/blob/8837557636432cd7c6202a67e5d7d2ad5fdd4db6/packages/reakit/src/Tabbable/Tabbable.ts#L98

This is primarily used to ensure that focus works consistently across browsers (for example, some browsers put focus on buttons when you click on them, others don't). But this is also preventing text selection unfortunately. There should be a better way to ensure focus consistency other than preventing the default mouse down behavior.

As a workaround, you can filter out the `onMouseDown` prop: https://codesandbox.io/s/reakit-rover-text-selection-wvuyi

This is related to #432 and #445

**Comment by diegohaz** [comment]
The "best" solution I can think of is rendering `button`s as `div`s by default and remove this whole `onMouseDown` interception.

> When in doubt, non-interactive elements like div and span have more consistent support for tabindex than interactive ones like a and button, despite the fact that the attribute is technically not valid when used with them according to the specification. If an element absolutely needs to be in the document’s tab order cross-browser without requiring specific system preferences or keyboard modifiers, div might be your best choice. Just be sure to add appropriate ARIA attributes and keyboard event handlers so that users can tell what the element is and interact with it when they get to it.
>
> — https://www.alexlande.com/articles/cross-browser-tabindex-woes/

**Comment by Slapbox** [comment]
Ahhh I looked through `Rover` for clues, but didn't make it over to `Tabbable`. Good catch by you as well and thank you!

Your solution works great and can actually be simplified down to simply wrapping the children with `{props => ()}`, but I don't understand what's going on. Can you help me understand why this works?

https://codesandbox.io/s/reakit-rover-text-selection-3lr8c

I'm not the best to comment on accessibility best practices, but `div`s sure seem a lot more predictable in a lot more situations in my experience.

**Comment by diegohaz** [comment]
Your example is not working. You are simply rendering a `span`. You have to pass `props` to it (filtering out `onMouseDown`).

**Comment by Slapbox** [comment]
Ah yes, I see. I'm using `Rover` as a replacement for `Step` which didn't have hotkey handling, and noticed I had my Rover component wrapped in a custom hotkey handler for arrow-based navigation from back in `v0`.

I technically have my use case working, but I'm trying to move to a 100% Reakit solution so possibly some more questions.

Thanks always @diegohaz!

**Comment by diegohaz** [comment]
Regarding the issue with `onMouseDown`, I posted a Twitter thread (putting it here for reference): https://twitter.com/diegohaz/status/1176998100495491072

**Comment by Slapbox** [comment]
Nestable buttons is a compelling reason. My first thought was, "What's the use case for nesting buttons?" - then I realized we have one. Not all of our application is done with Reakit (yet.)

If you're really torn on the matter, I read the thread and think that moving to `div` based buttons sounds preferable.

Your [comment on the downsides](https://twitter.com/diegohaz/status/1177000979704811521) sounds mostly to be downsides relates primarily to increased difficulty in development _of_ Reakit, not development _with_ Reakit, right? Not that I want to sign you up for extra work, but a lot of the time I hold off and try to make my original solution work I find I should have ditched it a while ago. I'm definitely not saying I know if this is one of those situations.

**Comment by jperasmus** [comment]
@diegohaz 

I'm not 100% sure about the history and discussions that led to the `onMouseDown` interception in the `Tabbable` component, so excuse if this is not relevant. From what I can gather, this interception was added to ensure cross-browser consistency, specifically for handling the focus state on button clicks (Firefox and Safari). I recently read an article that could offer an alternative solution to this problem, so that the `onMouseDown` interception code can be removed.

**Possible solution:**
- In the `Provider` component, add a global `click` event handler on the `document`. Inside the event handler callback, check if the `event.target` is a button (and also whether it is not currently the focused element) and then programmatically set focus to it. 
- Remove the `onMouseDown` interception code

This should ensure that Firefox and Safari both set the correct focus state on button clicks, while then still allowing the `onMouseDown` event to do its thing, like allow text selection, etc.

**Comment by diegohaz** [comment]
It has been fixed on #458. Will be available in the next release.
